test/extensions/filters/http/proto_api_scrubber/filter_config_test.cc

```diff
@@ -0,0 +1,484 @@
+                            }
+                          }
+                          custom_match: {
+                            typed_config: {
+                              [type.googleapis.com/xds.type.matcher.v3.CelMatcher] {
+                                expr_match: {
+                                  cel_expr_parsed: {
+                                    expr: {
+                                      id: 1
+                                      const_expr: {
+                                        bool_value: true
+                                      }
+                                    }
+                                    source_info: {
+                                      syntax_version: "cel1"
+                                      location: "inline_expression"
+                                      positions: {
+                                        key: 1
+                                        value: 0
+                                      }
+                                    }
+                                  }
+                                }
+                              }
+                            }
+                          }
+                        }
+                      }
+                      on_match: {
+                        action: {
+                          typed_config: {
+                            [type.googleapis.com/envoy.extensions.filters.http.proto_api_scrubber.v3.RemoveFieldAction] { }
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    )pb",
+                                                 GetParam().input_type_url);
+  ASSERT_TRUE(Protobuf::TextFormat::ParseFromString(proto_config_str, &proto_config));
+  NiceMock<Server::Configuration::MockFactoryContext> factory_context;
+  if (GetParam().exception_expected) {
+    try {
+      absl::StatusOr<std::shared_ptr<ProtoApiScrubberFilterConfig>> filter_config =
+          ProtoApiScrubberFilterConfig::create(proto_config, factory_context);
+      FAIL() << "Expected exception to be thrown.";
+    } catch (EnvoyException& ex) {
+      EXPECT_EQ(ex.what(), GetParam().expected_exception_message.value());
+    } catch (...) {
+      FAIL() << "Expected EnvoyException to be thrown but some other kind of Exception was thrown.";
+    }
+  } else {
+    absl::StatusOr<std::shared_ptr<ProtoApiScrubberFilterConfig>> filter_config =
+        ProtoApiScrubberFilterConfig::create(proto_config, factory_context);
+    EXPECT_EQ(filter_config.status().code(), GetParam().expected_status_code.value());
+    EXPECT_EQ(filter_config.status().message(), GetParam().expected_status_message.value());
+  }
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    MatcherInputTypeValidationTestSuite, MatcherInputTypeValidation,
+    testing::ValuesIn<MatcherInputTypeValidationTestCase>({
```

Comment:
I think this is a case where the tests should not be parameterized, as some of the parameters that are passed are unused (dependent on the test), and the tests become harder to read and maintain.
Please use non-parameterized tests here.